File: RepoTransBench/target_projects/Python/Go/stjordanis_betterprompt/tests/original/betterprompt_coverage_test.go
package original

import (
	"math"
	"testing"

	"betterprompt"
)

func sum(arr []float64) float64 {
	s := 0.0
	for _, v := range arr {
		s += v
	}
	return s
}

func TestCalculatePerplexityAllNone(t *testing.T) {
	// Simulate None in Python with sentinel value -100
	tokenLogprobs := []float64{-100, -100}
	result := betterprompt.CalculatePerplexity(tokenLogprobs)
	if !math.IsFinite(result) {
		t.Errorf("Expected finite, got: %v", result)
	}
}

func TestCalculatePerplexityRegular(t *testing.T) {
	tokenLogprobs := []float64{0, -1, -2}
	expected := math.Exp(-sum(tokenLogprobs) / float64(len(tokenLogprobs)))
	actual := betterprompt.CalculatePerplexity(tokenLogprobs)
	if math.Abs(actual-expected) >= 1e-8 {
		t.Errorf("Expected %v, got %v", expected, actual)
	}
}

func TestCalculatePerplexityEmptyList(t *testing.T) {
	result := betterprompt.CalculatePerplexity([]float64{})
	if !math.IsInf(result, 1) && !math.IsInf(result, -1) {
		t.Errorf("Expected Inf for empty input, got: %v", result)
	}
}

func TestCalculatePerplexityNaN(t *testing.T) {
	result := math.Exp(math.NaN())
	if !math.IsNaN(result) {
		t.Errorf("Expected NaN, got: %v", result)
	}
}

File: RepoTransBench/target_projects/Python/Java/stjordanis_betterprompt/src/test/java/com/betterprompt/original/BetterPromptCoverageTest.java
package com.betterprompt.original;

import com.betterprompt.BetterPrompt;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.*;

public class BetterPromptCoverageTest {
    @Test
    public void testCalculatePerplexityAllNone() {
        // Matching Python logic: [None, None] replaced with [-100.0, -100.0]
        List<Double> tokenLogprobs = Arrays.asList(null, null);
        List<Double> cleanLogprobs = new ArrayList<>();
        for (Double x : tokenLogprobs) {
            cleanLogprobs.add(x != null ? x : -100.0);
        }
        double result = BetterPrompt.calculatePerplexity(cleanLogprobs);
        assertTrue(Double.isFinite(result));
    }

    @Test
    public void testCalculatePerplexityRegular() {
        List<Double> tokenLogprobs = Arrays.asList(0.0, -1.0, -2.0);
        double expected = Math.exp(-(0.0 + -1.0 + -2.0) / 3.0);
        double actual = BetterPrompt.calculatePerplexity(tokenLogprobs);
        assertTrue(Math.abs(actual - expected) < 1e-8, "Perplexity must match expected.");
    }

    @Test
    public void testCalculatePerplexityEmptyList() {
        double result = BetterPrompt.calculatePerplexity(Collections.emptyList());
        assertTrue(Double.isInfinite(result));
    }

    @Test
    public void testCalculatePerplexityNaN() {
        double result = Math.exp(Double.NaN);
        assertTrue(Double.isNaN(result));
    }
}